src/components/Home.js

```diff
@@ -11,22 +12,53 @@ const Home = () => {
     url += '&app_id=' + ID;
     url += '&app_key=' + KEY;
 
+    const [recipes, setRecipes] = useState([]);
+    const [recipeTitles, setRecipeTitles] = useState([]);
+    const [recipeImages, setRecipeImages] = useState([]);
+    const [recipeIngredients, setRecipeIngredients] = useState([]);
+    const [recipeLinks, setRecipeLinks] = useState([]);
+    const [recipeIndexes, setRecipeIndexes] = useState([]);
+    const [cards, setCards] = useState(false);
+
     const getRandomRecipes = async () => {
         const response = await fetch(url);
         const body = await response.json();
-        console.log(body);
+        const recipeData = body.hits.map(hit => hit.recipe);
+
+        setRecipes(recipeData);
+        setRecipeTitles(recipeData.map(recipe => recipe.label));
+        setRecipeImages(recipeData.map(recipe => recipe.images.REGULAR));
+        setRecipeIngredients(recipeData.map(recipe => recipe.ingredientLines));
+        setRecipeLinks(recipeData.map(recipe => recipe.shareAs));
+        setRecipeIndexes(generateIndexes(recipeData.length));
+        
+        showCards();
+    }
+
+    const generateIndexes = (length) => {
+        const randomNumbers = new Set();
+        while (randomNumbers.size < 3) {
+            const randomNumber = Math.floor(Math.random() * length);
+            randomNumbers.add(randomNumber);
+        }
+        return Array.from(randomNumbers);
     }
 
+
     const buildURL = () => {
         let queryString = query === '' ? '' : `&q=${query}`;
         let dietString = diet === 'none' ? '' : `&health=${diet}`;
         let cuisineString = cuisine === 'none' ? '' : `&cuisineType=${cuisine}`;
         url = `https://api.edamam.com/api/recipes/v2?type=public&app_id=${ID}&app_key=${KEY}&ingr=${ingredients}${queryString}${dietString}${cuisineString}&mealType=${mealType}&time=${time}&random=true`;
     }
 
-    const handleClick = () => {
+    const showCards = () => {
```

Comment:
Why not just call setCards() directly?

Reply:
Since I also need to change the button title, this is no longer an issue.